src/handler.rs

```diff
@@ -199,6 +201,20 @@ where
         Ok(())
     }
 
+    #[inline]
+    fn eip7623_check_gas_floor(
+        &self,
+        evm: &mut Self::Evm,
+        exec_result: &mut <<Self::Evm as EvmTr>::Frame as FrameTr>::FrameResult,
+        init_and_floor_gas: InitialAndFloorGas,
+    ) {
+        // skip floor gas check for l1 messages.
+        if evm.ctx().tx().is_l1_msg() {
```

Comment:
Technically, l2geth does check the floor gas, it's just that it does not deduct it. But skipping the check is fine -- `L1MessageQueueV2` ensures that `gasLimit` is always large enough to cover `floorGas`.